modules_core/com.smf.jobs.defaults/src/com/smf/jobs/defaults/invoices/CreateFromOrder.java

```diff
@@ -108,16 +117,53 @@ protected ActionResult action(JSONObject parameters, MutableBoolean isStopped) {
       OBDal.getInstance().rollbackAndClose();
     }
 
+    ResponseActionsBuilder.MessageType messageType;
+    if (result.getType() == Result.Type.ERROR) {
+      messageType = ResponseActionsBuilder.MessageType.ERROR;
+    } else {
+      messageType = ResponseActionsBuilder.MessageType.SUCCESS;
+    }
+
     result.setResponseActionsBuilder(
         getResponseBuilder()
             .retryExecution()
             .refreshGridParameter(ORDER_GRID_PARAM)
-            .showMsgInProcessView(ResponseActionsBuilder.MessageType.SUCCESS, result.getMessage())
+            .showMsgInProcessView(messageType, result.getMessage())
     );
 
     return result;
   }
 
+  /**
+   * Checks if the BusinessPartner associated with an order specified in the given parameters has an invoice address.
+   *
+   * @param parameters
+   *     A JSONObject containing order details, including an order ID.
+   * @return true if the BusinessPartner associated with the order has an invoice address; false otherwise.
+   * @throws JSONException
+   *     If there is an error parsing the JSON data.
+   */
+  private boolean checkBusinessPartnerInvoiceAddress(JSONObject parameters) throws JSONException {
+    String orderId = parameters.getJSONObject(ORDER_GRID_PARAM).getJSONArray("_selection").getJSONObject(0).getString(
```

Comment:
This statement is not null-safe because it chains multiple method calls (getJSONObject, getJSONArray, getJSONObject, getString) without checking whether the intermediate objects are null before calling the next method in the chain. I would do something like:

```
private boolean checkBusinessPartnerInvoiceAddress(JSONObject parameters) throws JSONException {
    String orderId = getOrderIdFromParameters(parameters);
    Order order = OBDal.getInstance().get(Order.class, orderId);
    if (order == null) {
      throw new OBException(OBMessageUtils.messageBD("OrderNotFound"));
    }
    return hasInvoiceAddress(order.getBusinessPartner());
}

/**
 * Safely retrieves the order ID from the provided JSON parameters.
 *
 * This method checks for the existence of the necessary attributes in the JSON structure.
 * If any of the required fields are missing or null, it throws an OBException with a descriptive message.
 *
 * @param parameters The JSON object containing the parameters.
 * @return The order ID as a string.
 * @throws JSONException If any JSON processing errors occur.
 * @throws OBException If a required attribute is missing or null.
 */
private String getOrderIdFromParameters(JSONObject parameters) throws JSONException {
    if (parameters == null || !parameters.has(ORDER_GRID_PARAM)) {
        throw new OBException(OBMessageUtils.getI18NMessage("MissingAttribute", new String[]{ ORDER_GRID_PARAM }));
    }

    JSONObject orderGrid = parameters.getJSONObject(ORDER_GRID_PARAM);
    if (orderGrid == null || !orderGrid.has("_selection")) {
        throw new OBException(OBMessageUtils.getI18NMessage("MissingAttribute", new String[]{ "_selection" }));
    }

    JSONArray selectionArray = orderGrid.getJSONArray("_selection");
    if (selectionArray == null || selectionArray.length() == 0) {
        throw new OBException(OBMessageUtils.getI18NMessage("MissingAttribute", new String[]{ "_selection" }));
    }

    JSONObject firstSelection = selectionArray.getJSONObject(0);
    if (firstSelection == null || !firstSelection.has("id")) {
        throw new OBException(OBMessageUtils.getI18NMessage("MissingAttribute", new String[]{ "id" }));
    }

    return firstSelection.getString("id");
}
```

For this to work it is necessary to create a single message in the database, this message will receive a single parameter, here is an example screenshot:

![image](https://github.com/user-attachments/assets/ccf06b29-1f75-451a-85ab-6a4bcba06e2c)